extension/content/components/recipes/details/DetailsHeader.js

```diff
@@ -11,18 +11,32 @@ import {
   useRecipeDetailsData,
   useRecipeDetailsDispatch,
 } from "devtools/contexts/recipeDetails";
+import { actionIsPausable } from "devtools/utils/recipes";
 
 export default function DetailsHeader() {
   const data = useRecipeDetailsData();
   const dispatch = useRecipeDetailsDispatch();
+  // @ts-ignore
   const { recipeId, revisionId } = useParams();
   const {
     environment,
     selectedKey: environmentKey,
   } = useSelectedEnvironmentState();
   const history = useHistory();
   const normandyApi = useSelectedNormandyEnvironmentAPI();
-  const [isButtonLoading, setIsButtonLoading] = React.useState(false);
+  const [buttonsLoading, setButtonsLoading] = React.useState(new Set());
+
+  const addButtonLoading = (buttonName) => {
+    const newLoading = new Set(buttonsLoading);
+    newLoading.add(buttonName);
+    setButtonsLoading(newLoading);
+  };
+
+  const removeButtonLoading = (buttonName) => {
+    const newLoading = new Set(buttonsLoading);
+    newLoading.delete(buttonName);
+    setButtonsLoading(newLoading);
+  };
```

Comment:
I think this pattern is something we could probably re-use in the future: a hook that provides a set and methods to add and remove objects in a way that makes the hook mechanics happy.